doc/source/toolchain.rst

```diff
@@ -0,0 +1,272 @@
+
+C18 is a bug fix for C11, so C11 may be skipped entirely.
+
+
+In practice the C++ feature set that can be used is limited by the
+availability in the MS VisualStudio versions that SciPy needs to support.
+C++11 can be used, C++14/17 is going to be impossible
+for a very long time because of ecosystem support restrictions. See [4]_.
+
+.. note::
+
+    Developer Note: Some C99 features would be useful for scientific
+    programming, in particular better support of IEEE 754 [5]_.
+    SciPy has a small include file ``scipy/_lib/_c99compat.h`` which
+    provides access to a few functions.  Use in conjunction
+    with ``<numpy/npy_math.h>``.
+
+    ========================================= ========================================================
+     Feature                                  Workaround
+    ========================================= ========================================================
+    ``isnan()``, ``isinf()``, ``isfinite()``  Use ``sc_isnan()``, ``sc_isinf()``, ``sc_isfinite()``
+    ``NAN``                                   Use ``NPY_NAN`` (it is *almost* equivalent)
+    inline functions                          Make static functions and place in an include .h file
+    mid-block variable declarations           Declare variables at the top of the block
+    ========================================= ========================================================
+
+
+Fortran Compilers
+^^^^^^^^^^^^^^^^^
+
+Generally, any well-maintained compiler is likely suitable and can be
+used to build SciPy.
+
+======== ==================
+ Tool     Version
+======== ==================
+gfortran   >= 4.8.0
+ifort     A recent version
+flang     A recent version
+======== ==================
+
+
+Cython Compiler
+^^^^^^^^^^^^^^^
+
+SciPy always requires a recent Cython compiler.
+
+======== ============ ===============
+ Tool    Tool Version  SciPy version
+======== ============ ===============
+Cython     >= 0.29.2   1.2.1
+======== ============ ===============
+
+
+
+Other Libraries
+^^^^^^^^^^^^^^^
+
+Any library conforming to the BLAS/LAPACK interface may be used.
+OpenBLAS, ATLAS, MKL, BLIS and reference Netlib libraries are known to work.
+
+=============== =====================================================
+ Library           Minimum version
+=============== =====================================================
+LAPACK           3.4.1
+BLAS             A recent version of OpenBLAS, MKL or ATLAS.
+                 The Accelerate BLAS is no longer supported.
+=============== =====================================================
+
+
+There are some additional optional dependencies.
+
+=============== ======== ==========================================
+ Library        Version   URL
+=============== ======== ==========================================
+mpmath          Recent    http://mpmath.org/
+scikit-umfpack  Recent    https://pypi.org/project/scikit-umfpack/
+=============== ======== ==========================================
```

Comment:
@rgommers should I add scikit-sparse?

Reply:
Probably a good idea. Actually, could you run it by the mailing list first? We didn't have any discussion on this. My impression is that it should be fine, however for an indirect GPL dependency it's always good to have a few more eyes ....